acceptance/openstack/containerinfra/v1/containerinfra.go

```diff
@@ -0,0 +1,92 @@
+package v1
+
+import (
+	"testing"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/acceptance/clients"
+	"github.com/gophercloud/gophercloud/acceptance/tools"
+	"github.com/gophercloud/gophercloud/openstack/containerinfra/v1/clustertemplates"
+	th "github.com/gophercloud/gophercloud/testhelper"
+)
+
+// CreateClusterTemplate will create a random cluster tempalte. An error will be returned if the
+// cluster-template could not be created.
+func CreateClusterTemplate(t *testing.T, client *gophercloud.ServiceClient) (*clustertemplates.ClusterTemplate, error) {
+	choices, err := clients.AcceptanceTestChoicesFromEnv()
+	if err != nil {
+		return nil, err
+	}
+
+	name := tools.RandomString("TESTACC-", 8)
+	t.Logf("Attempting to create cluster template: %s", name)
+
+	createOpts := clustertemplates.CreateOpts{
+		Name:                name,
+		Labels:              map[string]string{},
+		FixedSubnet:         "",
+		MasterFlavorID:      "",
+		NoProxy:             "10.0.0.0/8,172.0.0.0/8,192.0.0.0/8,localhost",
+		HTTPSProxy:          "http://10.164.177.169:8080",
+		TLSDisabled:         false,
+		KeyPairID:           "kp",
+		Public:              false,
+		HTTPProxy:           "http://10.164.177.169:8080",
+		DockerVolumeSize:    3,
+		ServerType:          "vm",
+		ExternalNetworkID:   choices.ExternalNetworkID,
+		ImageID:             choices.ImageID,
+		VolumeDriver:        "cinder",
+		RegistryEnabled:     false,
+		DockerStorageDriver: "devicemapper",
+		NetworkDriver:       "flannel",
+		FixedNetwork:        "",
+		COE:                 "kubernetes",
+		FlavorID:            "m1.small",
+		MasterLBEnabled:     true,
+		DNSNameServer:       "8.8.8.8",
+	}
```

Comment:
Are all of these options necessary to create a basic template? The install docs only show a few options needed: https://docs.openstack.org/magnum/latest/install/launch-instance.html#provision-a-docker-swarm-cluster-and-create-a-container

Reply:
Will probably need to discuss what is actually needed. From the code only see couple mandatory fields, but when ran with just those mandatory fields, it doen't work.

Reply:
That's really curious. This is what is being POST'd when running the command shown in the above doc link:

```json
{
    "coe": "swarm",
    "dns_nameserver": "8.8.8.8",
    "docker_storage_driver": "devicemapper",
    "docker_volume_size": null,
    "external_network_id": "public",
    "fixed_network": null,
    "fixed_subnet": null,
    "flavor_id": "m1.small",
    "http_proxy": null,
    "https_proxy": null,
    "image_id": "fedora-atomic-latest",
    "keypair_id": null,
    "labels": {},
    "master_flavor_id": "m1.small",
    "master_lb_enabled": false,
    "name": "swarm-cluster-template",
    "network_driver": null,
    "no_proxy": null,
    "public": false,
    "registry_enabled": false,
    "server_type": "vm",
    "tls_disabled": false,
    "volume_driver": null
}
```

The notable omissions are the IPs and proxy-related fields. 

What I would be curious about is if Magnum wants `null` to be explicitly passed or if the fields can be omitted. If the former, then that is going to be a unique behavior of Magnum (all OpenStack services have their quirks) and we'll need to account for that accordingly (ie: no `omitempty`).